yt_dlp/mp4_parser.py

```diff
@@ -0,0 +1,136 @@
+import struct
```

Comment:
Merge this file into the PP file. I don't think it needs to be here

Reply:
ALso, I have no clue how to even starting reading this. Could you add some docstrings?

Reply:
> Merge this file into the PP file. 

I think joining them would bloat up mp4direct.py though...

> Could you add some docstrings?

Why not